### ISSUE TITLE
standardize dcache bounds checking for links

### ISSUE THREAD

**Issue by mmcgee-jump** [firedancer-io/firedancer#826]
Since all links are now specified in the topology, we can automatically bounds check messages received on links (mtu, chunk, wmark etc) to make sure they are valid, rather than every tile needing to remember to do this in `during_frag` etc

**Comment by ameya-deshmukh** [comment]
@mmcgee-jump is this open to pick up?

**Comment by mmcgee-jump** [comment]
Sure! Although it's not obvious to me how to do it cleanly, and might be a tricky task. The mux tile kind of needs a way to know the extents of the dcache that each fragment it's receiving is for. It doesn't have that knowledge currently, and it might be tricky to provide it (eg, sometimes a fragment has passed through multiple mcaches, and mixed together into another mcache stream, so it would also need to understand how that).

**Comment by ripatel-fd** [comment]
@mmcgee-jump IMO we should standardize the use of the `orig` field. Then, pass a table of (chunk) bounds for each orig to the recipient.

**Comment by mmcgee-jump** [comment]
Not sure what you mean by orig field here? tsorig? If the producer can specify orig field they would just forge a bad one. The consumer needs to basically know where frags can come at init without being told at runtime.

**Comment by mulupuruvikas** [comment]
@mmcgee-jump I would like to contribute to this issue. Is it free to pick up?

**Comment by mmcgee-jump** [comment]
@mulupuruvikas Yup sure! Go ahead. We did get rid of the need for orig specification, since all tiles copy the data now the orig is just the sending tile

**Comment by mulupuruvikas** [comment]
@mmcgee-jump would you be available to schedule a meeting (my email is in my bio)? I want to start contributing more to Firedancer and I would love to build a solid foundation about the codebase so that I can efficiently contribute to issues.

**Comment by mulupuruvikas** [comment]
@mmcgee-jump I created [this PR](https://github.com/firedancer-io/firedancer/pull/5554) to fix this issue. I also had the following questions:
1. In the organization.txt file, the build/ directory is supposed to contain build artifacts and test binaries. This subdirectory does not exist anymore, and as a result I don't think I can execute the unit tests I added. Is there a workaround to this?
2. (Based off the the build and test issues mentioned above) How can I see the telemetry for how my dcache bounds check centralization  improves performance?

**Comment by ripatel-fd** [comment]
> In the organization.txt file, the build/ directory is supposed to contain build artifacts and test binaries. This subdirectory does not exist anymore, and as a result I don't think I can execute the unit tests I added. Is there a workaround to this?

This directory should get created when running `make -j unit-test`

**Comment by mulupuruvikas** [comment]
Makes sense, I updated the PR with the changes you requested. Let me know if this is in the right direction, and I'll drop in the test cases.

**Comment by mulupuruvikas** [comment]
@mmcgee-jump @ripatel-fd gentle push, can I get this checked out?

**Comment by Jotanune** [comment]
Saw the previous PR stalled, so I took a fresh stab at this. I moved the bounds pre-calculation to the stem init phase for zero-overhead validation, linking my PR below.
